tests/basic/pure_test.go

```diff
@@ -2166,3 +2166,197 @@ func faLUNExists(faVolList []string, pvc string) bool {
+									log.InfoD("Delete pod %v", pod.Name)
+
+									if currentPod.Name == pod.Name {
+										log.FailOnError(fmt.Errorf("Pod %v is still present", pod.Name), "Pod %v should be deleted", pod.Name)
+										return nil, true, nil
+									}
+								}
+								return nil, false, nil
+							}
+							_, err = task.DoRetryWithTimeout(t, 5*time.Minute, 30*time.Second)
+							log.FailOnError(err, "Failed to wait for pods to delete")
+						}
+					}(ctx)
+				}
+				wg.Wait()
+			})
+
+			stepLog = "run the multipath -ll command on the node where the pods were scheduled before deleting"
+			Step(stepLog, func() {
+				// sleep for 60 seconds for all the entries to update
+				time.Sleep(30 * time.Second)
+				log.InfoD("Sleeping for 30 seconds for all the entries to update")
+				cmd := fmt.Sprintf("multipath -ll")
+				output, err := runCmd(cmd, selectedNode)
+				log.FailOnError(err, "Failed to run multipath -ll command on node %v", selectedNode.Name)
+				log.InfoD("Output of multipath on provisioned node -ll command: %v", output)
+				//check if the device path is present in multipath
+				if !strings.Contains(output, "failed faulty running") {
+					log.FailOnError(fmt.Errorf("Multipath device error not detected"), "Multipath device error should be detected")
+				}
+
+				stepLog = "Check if pod is scheduled on other node and validate if the volume is attached on the new node"
+				Step(stepLog, func() {
+					pods, err := core.Instance().GetPods(contexts[0].App.NameSpace, nil)
+					log.FailOnError(err, "Failed to get pods in namespace %v", contexts[0].App.NameSpace)
+					for _, pod := range pods.Items {
+						log.InfoD("Pod name: %v, node name: %v", pod.Name, pod.Spec.NodeName)
+						if pod.Spec.NodeName != selectedNode.Name {
+							log.InfoD("Pod %v is scheduled on node %v", pod.Name, pod.Spec.NodeName)
+							break
+						}
+					}
+				})
+			})
+			stepLog = "Start portworx on the node where the volume was attached"
+			Step(stepLog, func() {
+				StartVolDriverAndWait([]node.Node{selectedNode})
+			})
+
+			stepLog = "Check if the old multipath device entry is deleted from the node where the volume was attached"
+			Step(stepLog, func() {
+				//sleep for some time for the entries to update
+				time.Sleep(30 * time.Second)
+				
+				//run the multipath -ll command on the node where the volume is attached
+				cmd := fmt.Sprintf("multipath -ll")
+				output, err := runCmd(cmd, selectedNode)
+				log.FailOnError(err, "Failed to run multipath -ll command on node %v", selectedNode.Name)
+				log.InfoD("Output of multipath -ll command: %v", output)
+				//check if the device path is present in multipath
+				for _, devicePath := range devicePaths {
+					if strings.Contains(output, devicePath) {
+						log.FailOnError(fmt.Errorf("Multipath device %v is still present", devicePath), "Multipath device %v should be deleted", devicePath)
+					}
+				}
+				output, err = runCmd(cmd, selectedNode)
```

Comment:
why are you running twice. you already got the output. check the same

Reply:
Done